### PR TITLE
Added some basic Safari Zone Achievements

### PR DESCRIPTION
**Pull request by Jaaslet**
## Description

Adds achievements for bait, rocks, steps and items in the safari zone.
Only added some very basic/easy to get, because there is a chance this will be released before friend safari is out, and it will be mean to force players to do a lot of safari zone stuff, before we finish the rework

## Motivation and Context

We need more stuff to make the safari zone fun, and i think achievements are easy to add, and give a small reward for doing stuff (and rewards are fun)

## How Has This Been Tested?

Did stuff in the safari zone and saw the achievement numbers go up

## Types of changes

- Improving existing feature

### CODE REVIEW

**Comment by Jaaslet** [comment]
The plan is to add more achievements right before we release friend safari

File: src/modules/requirements/SafariBaitRequirement.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    constructor(value: number, option: GameConstants.AchievementOption = GameConstants.AchievementOption.more) {
+        super(value, option, GameConstants.AchievementType.Safari);
+    }
+
+    public getProgress() {
+        return Math.min(App.game.statistics.safariBaitThrown(), this.requiredValue);
+    }
+
+    public hint(): string {
+        return `${this.requiredValue} Bait needs to be thrown in the Safari Zone.`;
```
Comment (Farboo171) [inline src/modules/requirements/SafariBaitRequirement.ts:14]:
```suggestion
        return `${this.requiredValue} Bait needs to be thrown in a Safari Zone.`;
```

File: src/modules/requirements/SafariItemsRequirement.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    constructor(value: number, option: GameConstants.AchievementOption = GameConstants.AchievementOption.more) {
+        super(value, option, GameConstants.AchievementType.Safari);
+    }
+
+    public getProgress() {
+        return Math.min(App.game.statistics.safariItemsObtained(), this.requiredValue);
+    }
+
+    public hint(): string {
+        return `${this.requiredValue} Items needs to be picked up in the Safari Zone.`;
```
Comment (Farboo171) [inline src/modules/requirements/SafariItemsRequirement.ts:14]:
```suggestion
        return `${this.requiredValue} Items needs to be picked up in a Safari Zone.`;
```

File: src/modules/requirements/SafariRocksRequirement.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    constructor(value: number, option: GameConstants.AchievementOption = GameConstants.AchievementOption.more) {
+        super(value, option, GameConstants.AchievementType.Safari);
+    }
+
+    public getProgress() {
+        return Math.min(App.game.statistics.safariRocksThrown(), this.requiredValue);
+    }
+
+    public hint(): string {
+        return `${this.requiredValue} Rocks needs to be thrown in the Safari Zone.`;
```
Comment (Farboo171) [inline src/modules/requirements/SafariRocksRequirement.ts:14]:
```suggestion
        return `${this.requiredValue} Rocks needs to be thrown in a Safari Zone.`;
```

File: src/modules/requirements/SafariStepsRequirement.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    constructor(value: number, option: GameConstants.AchievementOption = GameConstants.AchievementOption.more) {
+        super(value, option, GameConstants.AchievementType.Safari);
+    }
+
+    public getProgress() {
+        return Math.min(App.game.statistics.safariStepsTaken(), this.requiredValue);
+    }
+
+    public hint(): string {
+        return `${this.requiredValue} Steps needs to be taken in the Safari Zone.`;
```
Comment (Farboo171) [inline src/modules/requirements/SafariStepsRequirement.ts:14]:
```suggestion
        return `${this.requiredValue} Steps needs to be taken in a Safari Zone.`;
```

File: src/scripts/achievements/AchievementHandler.ts [hunk 5]
```diff
@@ -401,6 +401,11 @@ class AchievementHandler {
         AchievementHandler.addAchievement('Priest in Training', 'Purify 50 unique Shadow Pokémon', new ShadowPokemonRequirement(50, GameConstants.ShadowStatus.Purified), 6, GameConstants.ExtraAchievementCategories.orre);
         AchievementHandler.addAchievement('Hand of Light', 'Purify all 131 unique Shadow Pokémon', new ShadowPokemonRequirement(131, GameConstants.ShadowStatus.Purified), 12, GameConstants.ExtraAchievementCategories.orre);
 
+        AchievementHandler.addAchievement('Who dropped these here?', 'Pick up 10 Items in the Safari Zone', new SafariItemsRequirement(10), 0.2);
```
Comment (Farboo171) [inline src/scripts/achievements/AchievementHandler.ts:404]:
```suggestion
        AchievementHandler.addAchievement('Who dropped these here?', 'Pick up 10 Items in a Safari Zone', new SafariItemsRequirement(10), 0.2);
```

File: src/scripts/achievements/AchievementHandler.ts [hunk 6]
```diff
@@ -401,6 +401,11 @@ class AchievementHandler {
...
+        AchievementHandler.addAchievement('This is not a Pokéball!', 'Throw 10 Rocks in the Safari Zone', new SafariRocksRequirement(10), 0.2);
```
Comment (Farboo171) [inline src/scripts/achievements/AchievementHandler.ts:405]:
```suggestion
        AchievementHandler.addAchievement('This is not a Pokéball!', 'Throw 10 Rocks in a Safari Zone', new SafariRocksRequirement(10), 0.2);
```

File: src/scripts/achievements/AchievementHandler.ts [hunk 7]
```diff
@@ -401,6 +401,11 @@ class AchievementHandler {
...
+        AchievementHandler.addAchievement('Hope it likes berries', 'Throw 10 Bait in the Safari Zone', new SafariBaitRequirement(10), 0.2);
```
Comment (Farboo171) [inline src/scripts/achievements/AchievementHandler.ts:406]:
```suggestion
        AchievementHandler.addAchievement('Hope it likes berries', 'Throw 10 Bait in a Safari Zone', new SafariBaitRequirement(10), 0.2);
```

File: src/scripts/achievements/AchievementHandler.ts [hunk 8]
```diff
@@ -401,6 +401,11 @@ class AchievementHandler {
...
+        AchievementHandler.addAchievement('Gotta get your steps in!', 'Walk 100 Steps in the Safari Zone', new SafariStepsRequirement(100), 0.2);
```
Comment (Farboo171) [inline src/scripts/achievements/AchievementHandler.ts:407]:
```suggestion
        AchievementHandler.addAchievement('Gotta get your steps in!', 'Walk 100 Steps in a Safari Zone', new SafariStepsRequirement(100), 0.2);
```

**Review by Farboo171 — CHANGES_REQUESTED** [review]
Changed "the" to "a" to reflect that these may be achieved in any safari zone.

**Review by Aegyo — APPROVED** [review]

**Pull request merged**